lib/timeval.c

```diff
@@ -60,7 +60,26 @@ struct curltime Curl_now(void)
   struct timeval now;
   struct curltime cnow;
   struct timespec tsnow;
-  if(0 == clock_gettime(CLOCK_MONOTONIC, &tsnow)) {
+
+  /*
+  ** clock_gettime() may be defined by Apple's SDK as weak symbol thus 
+  ** code compiles but fails during run-time if clock_gettime() is 
+  ** called on unsupported OS version.
+  */
+#if defined(__APPLE__) && defined(__clang__) && defined(__has_attribute)
+#if __has_attribute(availability)
+#define HAVE_CLOCK_GETTIME_CHECK
+  static int have_clock_gettime = 0;
+  if(__builtin_available(macOS 10.12, iOS 10, tvOS 10, watchOS 3, *))
+      have_clock_gettime = 1;
```

Comment:
Why 'static'  and shouldn't we rather use 'bool' (for readability)? The check seems to be done always, right?

And the check "__builtin_available(macOS 10.12, iOS 10, tvOS 10, watchOS 3, *))" just checks for certain OS versions, right? So is there documentation somewhere that specifies that `clock_gettime` is only available as a weak symbol starting in those versions?

Reply:
Sorry, I did not notice usage of ```bool``` type in curl, will adjust and commit the correction!

The ```time.h``` of libc source code can be used as documentation which has declaration for ```clock_gettime```, [here is the reference to it](http://opensource.apple.com/source/Libc/Libc-1244.30.3/include/time.h.auto.html), it is line 195 
```c
#define __CLOCK_AVAILABILITY __OSX_AVAILABLE(10.12) __IOS_AVAILABLE(10.0) __TVOS_AVAILABLE(10.0) __WATCHOS_AVAILABLE(3.0)
```

As you can see ```__IOS_AVAILABLE``` defines availability for macOS 10.12, iOS 10, tvOS 10, watchOS 3.